### ISSUE TITLE
Panic "called `Option::unwrap()` on a `None` value"

### ISSUE THREAD

**Issue by bjorn3** [ozkriff/zoc#256]
Cross compiled from Bash on Ubuntu on Windows to i686-pc-windows-gnu.

Will try to reproduce on real linux.

Binary:
https://1drv.ms/u/s!ArN-vh3Q5uflg3jCDUgWI3r2ufX0

Backtrace:
```
thread 'main' panicked at 'called `Option::unwrap()` on a `None` value', ../src/libcore\option.rs:323
stack backtrace:
   0:   0xaf6e6e - <unknown>
   1:   0xaf7468 - <unknown>
   2: 0x77a05c0b - <unknown>
   3: 0x77a05b31 - <unknown>
   4:   0xb0831c - <unknown>
   5:   0x490cd9 - <unknown>
   6:   0x585528 - <unknown>
   7:   0x5863b8 - <unknown>
   8:   0x5876f7 - <unknown>
   9:   0x586471 - <unknown>
  10:   0x5876f7 - <unknown>
  11:   0x5878e5 - <unknown>
  12:   0x5883e9 - <unknown>
  13:   0x568d25 - <unknown>
  14:   0x568c5c - <unknown>
  15:   0x5943fd - <unknown>
  16:   0xaf7187 - <unknown>
  17:   0x5a6b32 - <unknown>
  18:   0x4013dd - <unknown>
  19: 0x768462c3 - <unknown>
  20: 0x77a20fd8 - <unknown>
```

**Comment by ozkriff** [comment]
I don't have win10 to reproduce this :(

What actions lead to the crash?

**Comment by bjorn3** [comment]
Just playing a few minutes.

**Comment by atallahade** [comment]
I have the same issue on debian testing.

Here is my stack backtrace:
```
thread 'main' panicked at 'called `Option::unwrap()` on a `None` value', ../src/libcore/option.rs:323
stack backtrace:
   1:     0x5610dc7638da - std::sys::imp::backtrace::tracing::imp::write::h917062bce4ff48c3
                        at /buildslave/rust-buildbot/slave/stable-dist-rustc-linux/build/obj/../src/libstd/sys/unix/backtrace/tracing/gcc_s.rs:42
   2:     0x5610dc7679ef - std::panicking::default_hook::{{closure}}::h0bacac31b5ed1870
                        at /buildslave/rust-buildbot/slave/stable-dist-rustc-linux/build/obj/../src/libstd/panicking.rs:247
   3:     0x5610dc765d66 - std::panicking::default_hook::h5897799da33ece67
                        at /buildslave/rust-buildbot/slave/stable-dist-rustc-linux/build/obj/../src/libstd/panicking.rs:263
   4:     0x5610dc7663b7 - std::panicking::rust_panic_with_hook::h109e116a3a861224
                        at /buildslave/rust-buildbot/slave/stable-dist-rustc-linux/build/obj/../src/libstd/panicking.rs:451
   5:     0x5610dc766244 - std::panicking::begin_panic::hbb38be1379e09df0
                        at /buildslave/rust-buildbot/slave/stable-dist-rustc-linux/build/obj/../src/libstd/panicking.rs:413
   6:     0x5610dc766169 - std::panicking::begin_panic_fmt::h26713cea9bce3ab0
                        at /buildslave/rust-buildbot/slave/stable-dist-rustc-linux/build/obj/../src/libstd/panicking.rs:397
   7:     0x5610dc7660f7 - rust_begin_unwind
                        at /buildslave/rust-buildbot/slave/stable-dist-rustc-linux/build/obj/../src/libstd/panicking.rs:373
   8:     0x5610dc7a2fad - core::panicking::panic_fmt::hcfbb59eeb7f27f75
                        at /buildslave/rust-buildbot/slave/stable-dist-rustc-linux/build/obj/../src/libcore/panicking.rs:69
   9:     0x5610dc7a2ee4 - core::panicking::panic::hd383cb12a44b01ff
                        at /buildslave/rust-buildbot/slave/stable-dist-rustc-linux/build/obj/../src/libcore/panicking.rs:49
  10:     0x5610dc459236 - <core::option::Option<T>>::unwrap::h8a98df46e8fa10a6
                        at /buildslave/rust-buildbot/slave/stable-dist-rustc-linux/build/obj/../src/libcore/macros.rs:21
  11:     0x5610dc4b58d1 - core::game_state::State::unit::he2489f0eeb189b50
                        at /[..]/zoc/core/src/game_state.rs:249
  12:     0x5610dc4c39d6 - core::fow::Fow::apply_event::hf9dd2c63f68ec705
                        at /[..]/zoc/core/src/fow.rs:156
  13:     0x5610dc4b751f - core::game_state::State::apply_event::h92bb7c7371430d63
                        at /[..]/zoc/core/src/game_state.rs:513
  14:     0x5610dc430c16 - zoc::tactical_screen::TacticalScreen::end_event_visualization::h0fb95cdc76cbcb9c
                        at /[..]/zoc/src/tactical_screen.rs:1008
  15:     0x5610dc431062 - zoc::tactical_screen::TacticalScreen::logic::h1beba4acc4febc2d
                        at /[..]/zoc/src/tactical_screen.rs:1035
  16:     0x5610dc431ce1 - <zoc::tactical_screen::TacticalScreen as zoc::screen::Screen>::tick::hdd8fe18e0b478a55
                        at /[..]/zoc/src/tactical_screen.rs:1142
  17:     0x5610dc410231 - zoc::visualizer::Visualizer::draw::h18d260e0e42613e9
                        at /[..]/zoc/src/visualizer.rs:60
  18:     0x5610dc41015c - zoc::visualizer::Visualizer::tick::h52e60ab5d2ff0b44
                        at /[..]/zoc/src/visualizer.rs:50
  19:     0x5610dc43f5a8 - zoc::main::hd48e3ada2d160f15
                        at /[..]/zoc/src/main.rs:54
  20:     0x5610dc76f4ca - __rust_maybe_catch_panic
                        at /buildslave/rust-buildbot/slave/stable-dist-rustc-linux/build/obj/../src/libpanic_unwind/lib.rs:97
  21:     0x5610dc766dd5 - std::rt::lang_start::hd661476ce2fc2931
                        at /buildslave/rust-buildbot/slave/stable-dist-rustc-linux/build/obj/../src/libstd/panicking.rs:332
                        at /buildslave/rust-buildbot/slave/stable-dist-rustc-linux/build/obj/../src/libstd/panic.rs:351
                        at /buildslave/rust-buildbot/slave/stable-dist-rustc-linux/build/obj/../src/libstd/rt.rs:57
  22:     0x5610dc44c642 - main
  23:     0x7f04c19912b0 - __libc_start_main
  24:     0x5610dc2b9f29 - _start
  25:                0x0 - <unknown>
```

Compiling with rustc 1.14.0 (e8a012324 2016-12-16).

**Comment by ozkriff** [comment]
Closed as duplicate of #239
